Automated Migrations: Adapt the Instructions step for the Automated Migration flow

valterlorran opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Closes #93638

## Proposed Changes

* Add the new copies to the Instructions step to reflect the changes in phase one.
* I also fixed a few issues:
  * The sizes of the action items could cause a visual bug.
  * There was a ` char when it should '.
* When the user skips the credentials step, we should show the current design of the Instruction step.
* This PR also prevents the request to create the ticket when we already created it on the credentials step.

![Captura de Tela 2024-08-20 às 06 50 49](https://github.com/user-attachments/assets/6c12ae95-4a63-49d2-b6c0-2b111429fb74)

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* As part of the Automated Migration, we are already collecting the credentials, and the instructions step should not say that we'd give instructions on how to share the credentials.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Apply this branch to your local environment or add the Calypso Live link below.
* Open the dev console and activate the feature flag by running the following command:
```
window.sessionStorage.setItem('flags', 'automated-migration/collect-credentials')
```
* Now go through the migration flow by navigating to `/start`:
  * Go through the domain selection step
  * Select the free plan on the `Plans` page
  * On the `Goals` step, select the "Import existing content or website" option and click continue
  * Input the source site URL in the Identify step
  * Next, select the "Migrate Site" option on the `Import or Migrate` step
  * Select "Do it for me" on the `How to migrate` step
  * Go through the checkout
  * Once you complete the checkout, you should land on the new `Credentials` step
  * Fill in the form and click on continue
  * Now, on the Instructions step, make sure the copy matches the Figma design: F8foKv974Gc5dTHSh0i0Wo-fi-257_2924

Note: you should also test skipping the Credentials step. The instructions step should not have change in this case.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

valterlorran committed:
Add the Automated Migration texts on the instructions step

valterlorran committed:
Add the skipping logic

valterlorran committed:
Add tests

merkushin reviewed (commented):
Works well. The copy is the same as in Figma. 

I only noticed that the icon for the second point is different:
![CleanShot 2024-08-20 at 00 27 19@2x](https://github.com/user-attachments/assets/474e0136-1c38-434a-a347-23320fef5816)

valterlorran committed:
Fix icon

valterlorran committed:
Prevent sending the ticket for the automated migration, the ticket will be generated on the Credentials step

valterlorran committed:
Add more tracking data

valterlorran commented:
@merkushin, I also added prevention to the ticket submission process for the automated migration flow. We should test it thoroughly to ensure we don't mess with the current behavior. I also added the missing dependencies to the useEffect.

andres-blanco reviewed (approved):
I've tested it and it's working correctly.
* I've tested without the flag and the ZD ticket is correctly created and appearing on the Migrate queue
* I've tested with the flag. The ZD ticket is not created and the screen matches the one on the Figma file

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16535425).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16535425).
